### ISSUE TITLE
Why the book is empty?

### ISSUE THREAD

**Issue by NiiightmareXD** [iced-rs/book#1]
is it only for me or the book is empty but when I go to iced.rs the get started button forward me to the empty link

**Comment by rdbo** [comment]
Everything at https://book.iced.rs is empty (except the introduction), and this repository mostly only contains empty `README.md`'s. I guess it's only a template, and the only resources for documentation are the API reference (https://docs.rs/iced/latest/iced/) and examples (https://github.com/iced-rs/iced/tree/master/examples)

**Comment by nemith** [comment]
Why publish an empty book?  This is a bad look for a library.  Better off to have nothing to have this.

**Comment by hecrj** [comment]
The book is empty because we have barely worked on it yet. For now, it's just a draft of the topics we'd like to write about.

The library is in very early stages and most of the features are largely unfinished (or not even started!).

Mostly, the unfinished book is there to:

- Let the existing community know that it is actively being worked on and where.
- Incentivize contributions, as they will go live as soon as they are merged.
- Discourage users that expect a production-ready library.

In other words, we are working on the book completely in the open. There is nothing to hide!

> Why publish an empty book? This is a bad look for a library. Better off to have nothing to have this.

@nemith Honestly, your comment is quite disrespectful. I hope you reflect on it before approaching other open-source maintainers.

**Comment by nemith** [comment]
@hecrj  I did not mean to be disrespectful but I spent some time trying to figure out if my browser wasn't rendering right.  It is much better to be clear that the book is coming later and to use the API generated docs and the example than to confuse new people, like me on what is going on.  

I think that it is turning people away from trying iced and may be counterproductive to have an empty book than no book. (until it is written of course or at least partial written).  

I disagree that an empty book fulfills any of your reasons for having besides discouraging users.  

> Let the existing community know that it is actively being worked on and where.

There was no indication that it was being worked on.  It was just a bunch of blank pages. I came to the github to make sure it wasn't my browser failing to render text all of sudden.  Also looking at the repository and seeing that there has been no activity for months doesn't feel like it's active at all.  

> Incentivize contributions, as they will go live as soon as they are merged.

I think that is good, but again there is no indication that any activity has been on the book besides a short into

> Discourage users that expect a production-ready library.

I think it does do a good job of discouraging users.  These users being discouraged could eventually become contributers to code or to the book.  


My comment wasn't to be disrespectful, but really look at having a better message about what is intended.  There isn't even a banner that states that the book is a work in progress.   It is literally a bunch of headings and empty pages.  I do think it would be better for the project to move the book to a "beta" part of your website and not presented when you click the "Getting Started" button.  

I am more than happy to do a PR here to add what you stated to the book so that at least people aren't wondering why none of the pages are rendering correctly.

**Comment by hecrj** [comment]
@nemith Independently of what your intention was. Your first comment was a passive-aggressive rhetorical question followed by very low effort, destructive criticism in absolute statements and no particular personal context. It reads: "Why would you do X? It's wrong and useless."

Now, your new comment is much better. Proper context and constructive criticism. Thank you!

And I agree with most of your points. We can fix most of the issues by adding a warning to the empty sections. It's something that I had in mind, but I haven't got around to it yet.

In any case, I don't think the website is getting much traffic since I haven't shared it anywhere yet. In fact, the website itself is quite unfinished too.

**Comment by nemith** [comment]
I can see how my first message could be seen passive aggressively and I apologize.  

> How did you find the website, if I may ask?

First hit when i DDG "iced gui"

**Comment by NiiightmareXD** [comment]
> @nemith Independently of what your intention was, your first comment was a passive-aggressive rhetorical question followed by very low effort, destructive criticism in absolute statements and no particular personal context. It reads: "Why would you do X? It's wrong and useless." You are talking about work that is being given away by other people!
> 
> Now, your new comment is much better. Proper context, constructive criticism, and a bit of empathy. Thank you!
> 
> And I agree with most of your points. We can fix most of the issues by adding a warning to the empty sections. It's something that I had in mind, but I haven't got around to it yet. Feel free to open that PR!
> 
> In any case, I don't think the website is getting much traffic since I haven't shared it anywhere yet. In fact, the website itself is quite unfinished too. How did you find the website, if I may ask?

I found it in the iced website

**Comment by PointyFluff** [comment]
You guys really need to work on the book and docs. 

Your library is nearly useless without them. 

Please!  Lack of docs are the major thing holding back rust GUIs!
